768 merge seqfetcher

reece opened the pull request:
This PR should close #768, #719, and #635.
When merged, we need to close #719 and #635 manually.
Please see comments regarding this commit in #768.

jeromekelleher commented on scripts/download_data.py:
```diff
@@ -116,6 +116,45 @@ def getMaxPos(self, chrom):
         return minMax.maxPos
 
 
+def fetchSequence(ac, startIndex=None, endIndex=None, as_list=False):
```
``as_list`` -> ``asList``. We need to camelCase all identifiers I'm afraid. (I'm looking forward to the change to protobuf so I won't need to say this any more!)

jeromekelleher replied:
Actually, why do we have the ``as_list`` argument here at all? We're only using this function in one place, and it uses the non-default value for this. It would seem simpler to just return a list in all situations, and use ``"".join(fetchSequence('NP_056374.2',0,10))`` for the doctests.

jeromekelleher commented on scripts/download_data.py:
```diff
@@ -116,6 +116,45 @@ def getMaxPos(self, chrom):
+
+    """
+
+    url_fmt = ("http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?"
```
camelCase

jeromekelleher commented on scripts/download_data.py:
```diff
@@ -116,6 +116,45 @@ def getMaxPos(self, chrom):
+
+    url_fmt = ("http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?"
+               "db=nucleotide&id={ac}&rettype=fasta")
+    if (startIndex is None or endIndex is None):
```
Delete the brackets here, as they aren't necessary.

jeromekelleher commented:
Looks good, thanks @reece. I have some minor nits above. The ``asList`` argument does make it simpler to test the output in the doctests, but at the cost of making the client code a bit more complex (and, in a way, not actually tested at all).

reece commented:
Removed as_list entirely

url_fmt -> urlFmt (ugh -- an exemplar of fugly casing)

removed parens

RE: why have as_list? Because no one would expect a function called fetchSequence to return a list of sequence snippets.

jeromekelleher commented:
Great, thanks @reece! I think we're good to merge once the commits are squashed.

I agree on the aesthetic stuff --- I am looking forward to having a snake_case codebase sometime soon!

jeromekelleher commented:
Can we squash the commits please @reece? It would be nice to merge this as @dcolligan has another patch ready to land on the same code in #778.

reece commented:
@jeromekelleher: collapsed, committed, and checks passed.

jeromekelleher commented:
Excellent, thanks @reece. +1, merging.
